quiz/get_quiz.py

```diff
@@ -103,7 +104,19 @@ def get_quiz_responses(json_data):
     responses = results.get('responses', [])
     # TODO Convert to json format needed for Item Analysis
     # Call Analyze_test and send analysis data in the below dict
-    quiz_analysis = {}
+    exam_info = {'exam': {'name': results.get('title')}, 'student_list': []}
+
+    for i in results.get('students'):
+        curr_id = str(i[0].get('student_id'))
+        exam_info['student_list'].append({'id': curr_id, 'item_responses': []})
+
+    for i in results.get('responses'):
+        for k in i:
```

Comment:
Instead of `responses` being a nested list (list of lists of objects), it can just be one list (list of objects) since each object has the item id. That way it can just be one loop instead of two.

Reply:
```
function getQuizResponses(form_url) {
  // Open a form by URL.
  var form = FormApp.openByUrl(form_url);

  items = []
  responses = []
  student_list = []
  var formResponses = form.getResponses();
  for (var i = 0; i < formResponses.length; i++) {
      var formResponse = formResponses[i];
      var itemResponses = formResponse.getItemResponses();
      students = []
      for (var j = 0; j < itemResponses.length; j++) {
        var itemResponse = itemResponses[j];
        var type = itemResponse.getItem().getType()
        var response =  itemResponse.getResponse()
        //Logger.info(typeof(response))
        
        if (type == 'MULTIPLE_CHOICE' || type == 'CHECKBOX') {
          
          var resp = {student_id: parseInt(i + 1), item_id: parseInt(j), score: 1, student_response:response}
          var item = {item_id: parseInt(j), item_text: itemResponse.getItem().getTitle(), item_type: type}
          var q_item = (type == 'MULTIPLE_CHOICE' ? itemResponse.getItem().asMultipleChoiceItem() : itemResponse.getItem().asCheckboxItem()); 
          var choices = q_item.getChoices()
          var points = q_item.getPoints();
          if (!points || points==0) continue;
          correct_answer = []
          for (var k = 0; k < choices.length; k++) {
            var correct_ans = choices[k].isCorrectAnswer()
            var ch = choices[k].getValue()
            if (correct_ans) correct_answer.push(ch)
            //Logger.info(ch)            
            //Logger.info(correct_ans)
            var correct = (typeof(response) === 'string' ? ch == response : response.indexOf(ch) >= 0) 
            //Logger.info(correct)
            if (correct !== correct_ans)
               resp['score'] = 0
          }
          item['correct_answer'] = correct_answer
          responses.push(resp)
          Logger.info(resp)  
          if (i==0) items.push(item)
        }
        else {
          var student = {student_id: parseInt(i + 1), student_response:response, text: itemResponse.getItem().getTitle(), item_type: type}        
          students.push(student)
          Logger.info(student) 
        }
      }
      student_list.push(students)
  }
  
  var file = DriveApp.getFileById(form.getId());
  var metadata = {
    id: form.getId(),
    published_url: form.getPublishedUrl(),
    editor_url: form.getEditUrl(),
    summary_url: form.getSummaryUrl(),
    updated: file.getLastUpdated().toJSON(),
    created: file.getDateCreated().toJSON()    
  }
  response = {
    title: form.getTitle(),
    description: form.getDescription(),
    responses_count: parseInt(formResponses.length), 
    responses: responses,
    students: student_list,
    items: items,
    metadata: metadata
  }  
  //Logger.log(responses)
  return(response)
  
}
```

This should work.

Reply:
There could be more than one student attributes like name, email, age, city etc.